### ISSUE TITLE
Getting rid of asteriks.

### ISSUE THREAD

**Issue by moskovskiy82** [felipecrs/dahua-vto-on-home-assistant#1]
Found a link for this solution on one of the telegram channels.  Is asterisk required only for the button pressed event?

As this is easily achieved with the dahua integration or i get something wrong? Would love to have the same set up but setting up an asterisk for this seems will take too much time

  trigger:
  - platform: state
    entity_id: binary_sensor.kalitka_vto_button_pressed
    from: 'off'
    to: 'on'

**Comment by felipecrs** [comment]
You can see in my automation.yaml that I already use such sensor:

https://github.com/felipecrs/dahua-vto-on-home-assistant/blob/164d142b7ee5e473e657e663848c2a140a8cd981/home-assistant/automations/doorbell-ringed.yaml#L4-L8

But on my tests, unless the VTO is connected to a SIP server (that's why Asterisk), such sensor won't fire.

- https://github.com/rroller/dahua/issues/359

**Comment by moskovskiy82** [comment]
Well strange as for me it works without any flow. I have a VTO3211D

Haven't almost touched any settings but if it helps here is the setup

![image](https://github.com/felipecrs/dahua-vto-on-home-assistant/assets/15857592/5941910b-9ca0-4594-98df-71bf78a5f35d)

![image](https://github.com/felipecrs/dahua-vto-on-home-assistant/assets/15857592/f9328f16-47a8-4f91-ace7-258f516e6919)

![image](https://github.com/felipecrs/dahua-vto-on-home-assistant/assets/15857592/804fe5ff-b5c7-4843-935b-a2d2572667cb)

![image](https://github.com/felipecrs/dahua-vto-on-home-assistant/assets/15857592/d63a21e8-e0cd-4a30-add3-b0a986d02513)

![image](https://github.com/felipecrs/dahua-vto-on-home-assistant/assets/15857592/c3c0f9d8-8364-44ce-a99d-b8b2e9a79d30)

![image](https://github.com/felipecrs/dahua-vto-on-home-assistant/assets/15857592/a64f0321-e285-44f8-9fa7-c988635d8972)
Seems here room no doesn't do ....t

UPnP enabled but all failed :) And everything works including DMSS

![image](https://github.com/felipecrs/dahua-vto-on-home-assistant/assets/15857592/44ddb922-2689-4890-9564-767b250c4a9a)

![image](https://github.com/felipecrs/dahua-vto-on-home-assistant/assets/15857592/7bb7d08b-1dc8-444a-b460-a4a6561edf2e)

![image](https://github.com/felipecrs/dahua-vto-on-home-assistant/assets/15857592/224cc2e8-5083-4dad-ac23-509429ade7e4)

**Comment by felipecrs** [comment]
Thank you. I had tested before using the VTO's own SIP Server and it didn't work. But I will test again. You are not the first one who says it works.

I have some questions though:

![image](https://github.com/felipecrs/dahua-vto-on-home-assistant/assets/29582865/9b203033-6eed-4403-85fd-151853c0bc38)

The IP address you blurred is the VTO's IP address?

> And everything works including DMSS

That is one difference. I purposely disabled cloud access on my VTO.

Also, can you please explain what happens when you press the doorbell button with this setup?

What audio does it play? If the sensor is working correctly, I would expect it to play "Calling now, please wait a moment" and then nothing else.

**Comment by moskovskiy82** [comment]
1. Yes VTO's. As you can see when the SIP Server check box is checked all options are not available for change. I guess IP is put up automatically.
2. As soon as the button is pressed "Calling now, please wait a moment" is played on VTO. DMSS shows an incoming call and the integration in HA fires an event. So i get an announcement "Someone is calling at the gate" on my smart speaker

**Comment by felipecrs** [comment]
> 1. Yes VTO's. As you can see when the SIP Server check box is checked all options are not available for change. I guess IP is put up automatically.

Yes, intuitively that's what it does, but I read somewhere that these options can be changed, but you need to change them with "Enable SIP Server" unticked, save, and then tick it. Not sure if it's true anyway.

> 2. As soon as the button is pressed "Calling now, please wait a moment" is played on VTO. DMSS shows an incoming call and the integration in HA fires an event. So i get an announcement "Someone is calling at the gate" on my smart speaker

Amazing. I wish I could replicate that. I will do some tests this weekend. I hope I don't need to enable cloud function for this to work though.

**Comment by felipecrs** [comment]
> followed by another announcement when the call fails (since I don't have anything set up to actually pick it up)

You can solve that by cancelling the call as soon as possible after it starts ringing. Check my automation:

https://github.com/felipecrs/dahua-vto-on-home-assistant/blob/d03427bc7475e5e718567aeee55f157722b20145/home-assistant/automations/doorbell-ringed.yaml#L13-L15

**Comment by felipecrs** [comment]
But hey guys, if the `button_pressed` sensor is working for you with the built-in SIP server, you don't need Asterisk. As mentioned in my setup, that's the only reason why I needed it.

You can still get inspiration from the rest. :D

**Comment by Cr4z33** [comment]
> Well strange as for me it works without any flow. I have a VTO3211D
> 
> Haven't almost touched any settings but if it helps here is the setup
Tell me what did you setup then in Home Assistant to be able to get the VTO calls?

Can you please extend the details?

I am trying to get this working without the need of Asterisk.

**Comment by Cr4z33** [comment]
OK while waiting for @moskovskiy82 reply I would like to give Asterisk a try.

Therefore @felipecrs can you please give me some further details? 🙏🏼

Like what "SIP Domain" did you enter to connect the VTO to Asterisk?

Did you leave "VDP" or did you enter "asterisk" (as suggested by someone in another issue I cannot remember about)?

Also what extension number is Home Assistant then? 801?

I've got by the way a VTO2202F-P-S2.

**Comment by felipecrs** [comment]
@Cr4z33 check the files in this repository, I believe everything is explained there.

**Comment by Cr4z33** [comment]
@felipecrs tell me did you have to enter the "5060" port  and the "admin" SIP server username in the Asterisk integration config?

Actually would you mind copying&pasteing the YAML config here please? 😬

I still can't get it working. 🥲

**Comment by Cr4z33** [comment]
@felipecrs thanks for sharing, but I got Asterisk working at last by editing `sip.conf`.

I had to change the default port to `5060`.

After that my VTO calls got accepted by Asterisk.

Now I have to keep adapting your Frigate card code to my entities and devices so it's still a long journey... 🥵

I hope you won't mind if I will reply here again?

At the end of the day I think it's useful for everybody interested into it. 😅

**Comment by felipecrs** [comment]
> I had to change the default port to 5060.

I think you messed up with something. You didn't need to do this. `sip.conf` is for the deprecated chan-sip support. And 5060 is already the default for pjsip (`pjsip.conf`).

Furthermore, chan-sip (`sip.conf`) comes even disabled by default unless you enable it on `modules.conf`. So... what you are describing makes no sense to me.

> I hope you won't mind if I will reply here again?

Absolutely not, please do it.

**Comment by Cr4z33** [comment]
Well before I was getting errors in the Asterisk logs when pressing the VTO button and also the Asterisk integration was accepting 5038 port only.

As soon as I changed the port to 5060 in sip.conf the integration accepted it too (and the errors were gone).

**Comment by Cr4z33** [comment]
@felipecrs so far I got FINALLY working the Frigate card! 😁

I am replying this just to leave it here as a note for everybody out there.

The main misunderstanding (if you are an average Home Assistant user like me hehe) in the Frigate Card code provided by Felipe is that `camera.doorbell` at line 6 is NOT a Home Assistant entity ID, but actually the name of one of the go2rtc streams entered in Felipe's Frigate [config](https://github.com/felipecrs/dahua-vto-on-home-assistant/blob/master/frigate/frigate.yaml) (see line 6)!

**Comment by felipecrs** [comment]
Hey @Cr4z33, in fact `camera.doorbell` is a Home Assistant entity. It's created by the Frigate integration for Home Assistant.

**Comment by Cr4z33** [comment]
> Hey @Cr4z33, in fact `camera.doorbell` is a Home Assistant entity. It's created by the Frigate integration for Home Assistant.

Oh even more at risk of misunderstanding then lol!

OK I take back what I said about go2rtc. 😅

**Comment by felipecrs** [comment]
But it's true, even though `camera.doorbell` is a true entity, I am still overriding its go2rtc stream URL in this line:

https://github.com/felipecrs/dahua-vto-on-home-assistant/blob/f824fb9271b2518b7250461a08dfca6e64aa899a/home-assistant/dashboard/doorbell.yaml#L11

**Comment by Cr4z33** [comment]
> But it's true, even though `camera.doorbell` is a true entity, I am still overriding its go2rtc stream URL in this line:
> 
> https://github.com/felipecrs/dahua-vto-on-home-assistant/blob/f824fb9271b2518b7250461a08dfca6e64aa899a/home-assistant/dashboard/doorbell.yaml#L11

Yeah I noticed it. 

Right now I am dealing with your ringing doorbell automation: first thing is that I replaced Fully Kiosk with Browser Mod that allows me to use more devices (ie. I've got a tablet running Linux), but generally speaking it's really a nice project for the Dahua VTOs! 👏🏼

Most of all I am happy I didn't have to setup a pfSense machineto deal with WebRTC SSL issues and others as needed with other projects.

It works even with the Amcrest AD410 video doorbell so I guess it might work with HikVision and IMOU (they all belong to the same company after all) too!

**Comment by felipecrs** [comment]
It should work with any camera that go2rtc can 2-way audio with. For Hikvision, you'll need @pergolafabio's [add-on](https://github.com/pergolafabio/Hikvision-Addons) to stop the call and receive the events.

**Comment by felipecrs** [comment]
A piece of news, Dahua released a new firmware for my VTO:

https://www.dahuasecurity.com/pt/products/All-Products/Video-Intercoms/IP-Series/Villa-Door-Station/Pro-Series/VTO2202F-P-S2

V4.600.0000000.2.R.240802

**Comment by convicte** [comment]
Just to reiterate @felipecrs , as I've managed to get SIP working on my house alarm panel - Dahua integration https://github.com/rroller/dahua is working just fine in detecting button presses without the SIP server running on the VTO.
Haven't looked into specific EVENTS though - to be honest.

I was treating over needing it to run SIP server on the VTO, since it didn't work with my Satel alarm system (it wouldn't recognize the VTO SIP), so was glad to see it works just fine with any SIP server setup thus far (more specifically mine is on INT-TSI panel from Satel).

![image](https://github.com/user-attachments/assets/d7a60e00-6b2a-4041-9309-6a2823f6e31d)

The issue I am having is I don't know if the VTO2101E-P-S1 (https://www.dahuasecurity.com/ceen/products/All-Products/Discontinued-Products/Video-Intercoms/VTO2101E-P-S1) has any chance of working with your setup, as the audio encoding doesn't seem to be adjustable (I believe).

The thing I would like to work is to have the option to pick up SIP connections from HA phone APP, as I have a SIP wall alarm panel serving as the permanent reception device. 
Any ideas how to easily get this part working for VTO which is unlikely to have ACC audio compression?
Help will be greatly appreciated.

**Comment by convicte** [comment]
FYI, Events are also being received:
![image](https://github.com/user-attachments/assets/9b971de9-6a4d-4f84-8b3d-87868c942671)

**Comment by felipecrs** [comment]
> The issue I am having is I don't know if the VTO2101E-P-S1 (https://www.dahuasecurity.com/ceen/products/All-Products/Discontinued-Products/Video-Intercoms/VTO2101E-P-S1) has any chance of working with your setup, as the audio encoding doesn't seem to be adjustable (I believe).

I advise you to test the solution to change the codec in [fix_vto_codecs.sh](https://github.com/felipecrs/dahua-vto-on-home-assistant/blob/master/go2rtc/fix_vto_codecs.sh) anyway. Or have you tried already and it doesn't work?

Otherwise, if you want to keep using SIP, then you can use the SIP hass card: https://github.com/TECH7Fox/sip-hass-card

BTW the purpose of this issue is to drop the dependency on an external SIP server. If I am understanding correctly, you still depend on one SIP server and that's why the Dahua integration works. If that's not the case, i.e. if you are using the VTO embedded SIP server, please clarify.

**Comment by convicte** [comment]
1) For me the situation is a bit more convoluted with the codec (and I must admit I didn't look into it yet), since my Frigate is running as a separate container under Unraid and not HA addon, so I am a bit lost where the script would have to copied into and how to set up privilages in the said location. If you can point me in the right direction, I'd love to test it out.
HA OS is running in a VM on the same Unraid server, btw.
2) I've tried the https://github.com/TECH7Fox/sip-hass-card but thusfar have not been able to figure it out - as there doesn't seem to be an option to provide a SIP server password, which would allow the card to connect. The documentation is very inconsistent as to how a password protected SIP server would be handled.
3) While I am using a SIP server to connect my VTO with a Satel INT-TSI wall pannel, this, I believe has nothing to do with the integration sending the button presses to HA. More specifically HA has no clue the SIP server exists and the SIP server is running on the wall pannel for the purpose of VTO to pannel connection. I hope that clarifies things.

Thanks!!

**Comment by felipecrs** [comment]
1. Copy the script to the same directory where you have `frigate.yaml`. Then, you can call it within Frigate/go2rtc using `/config/fix_vto_codecs.sh`.
2. 
![image](https://github.com/user-attachments/assets/05657f70-8b3b-4779-bdd2-d6231a618b32)
3. It's easy to test: power off your SIP server, reboot the VTO, and check if the button press event still works (it will not, I bet).

**Comment by convicte** [comment]
1) Thanks, I'll test it out, but it may need to wait until I am back from vacation. Looks simple enough, though! Will report here.
2) Sorry, that doesn't help. The secret is specific to a person entity, which I wouldn't have at all. Technically, I probably wouldn't be using extensions, but the card requires it.
If you have experience setting these up, how would you operate the following setup:
  - VTO call no. 8001
  - Wall panel no. 101 (i.e. Villa Call no.)
  - SIP server 192.168.1.7
  - port 5060
  - SIP password: 12345
  - VTO door open code: 22 (i.e. main lock)
  - VTO gate open code 4321 (i.e. second lock)
3) I am very curious so will check it out.

**Comment by felipecrs** [comment]
1. No worries. I am hopeful it will work though.
2. It has been a long time since I don't use the SIP Card. However, what I think is gonna be your problem is that the SIP Card can't connect directly to the SIP port. It needs to connect to the WSS port ([SIP over WebSocket](https://techdocs.audiocodes.com/session-border-controller-sbc/mediant-software-sbc/user-manual/version-740/content/um/SIP%20over%20WebSocket.htm)). Asterisk SIP Server supports it, but it's likely your SIP server doesn't.
   
   It's pointless trying to configure it unless you are sure your SIP server has WebSockets support.
3. :)

**Comment by convicte** [comment]
1) Thanks, as am I.
2) Yeah... that explains it... no way this server is supporting WSS... it's the absolute bare-bones setups for calling into a wall panel of an alarm system. So I can as well scratch the card all together.

Thanks for all the help!

**Comment by Cr4z33** [comment]
> 1. Thanks, I'll test it out, but it may need to wait until I am back from vacation. Looks simple enough, though! Will report here.
> 
>     2. Sorry, that doesn't help. The secret is specific to a person entity, which I wouldn't have at all. Technically, I probably wouldn't be using extensions, but the card requires it.
>        If you have experience setting these up, how would you operate the following setup:
> 
> 
>     * VTO call no. 8001
> 
>     * Wall panel no. 101 (i.e. Villa Call no.)
> 
>     * SIP server 192.168.1.7
> 
>     * port 5060
> 
>     * SIP password: 12345
> 
>     * VTO door open code: 22 (i.e. main lock)
> 
>     * VTO gate open code 4321 (i.e. second lock)
> 
> 
>     3. I am very curious so will check it out.

Keep us updated as I'm using Felipe's solution and I am trying to include also my VTH touch panel to get the VTO SIP calls (at the same time with the Asterisk server).

**Comment by Cr4z33** [comment]
@convicte so has there been any success? 🙂

**Comment by esand** [comment]
Came here to mention that Asterisk should hopefully not be necessary. I don't have a Dahua/Amcrest, but a similar Lorex and found a repo called DahuaVTO2MQTT that speaks the Dahua API and publishes messages to MQTT. Since Frigate usually means you've got an MQTT broker, this may be a simpler solution (if I understand correctly, the Dahua Home Assistant add-on is probably the same thing... so this is if you can't use add-ons, or if it isn't working for you for any other reason).

The main repo is here: https://gitlab.com/elad.bar/DahuaVTO2MQTT
I found that at one point, that version wasn't working for me so I had to use this one: https://github.com/baudneo/dahuavto2mqtt

Take your pick and hopefully one works for you (they're both docker containers). They should publish all the events you'll need - motion, button press, etc...

**Comment by felipecrs** [comment]
@esand, do you use a VTH?

**Comment by esand** [comment]
> @esand, do you use a VTH?

No, I do not. I only have the Lorex doorbell which is built using Dahua stuff and a tweaked API that the DahuaVTO2MQTT app seems to still work with enough that I can get doorbell button press events (all I need it for).

**Comment by felipecrs** [comment]
Got it. I'm very unoptimistic about it, considering https://github.com/rroller/dahua/issues/359, but I'll give it a try once I figure it out.

**Comment by pedromrg** [comment]
Got rid of Asterisk and found another trigger.

Used: https://github.com/arnonym/ha-plugins ha-sip

Config:

enabled: true
registrar_uri: sip:192.168.1.XX #VTO IP
id_uri: sip:9901#2@192.168.1.25 # used 99001#2 to connect to VTO ( #0 and #1 is used for my 2 VTH's)
realm: "*"
user_name: 9901#2 # as mentioned above
password: "123456" # default password 
answer_mode: listen
settle_time: 1
incoming_call_file: ""

Now ha-sip is connected to the SIP Server running inside the VTO hence my HA now has an extension number.

To trigger the automation I used a webhock trigger, creating an automation with the webhook trigger and adding the webhook ID generated on the automation to the ha-sip config:

webhook:
    id: sip_call_webhook_id

[https://github.com/arnonym/ha-plugins ha-sip](https://github.com/arnonym/ha-plugins?tab=readme-ov-file#incoming-calls)

I was forced to find this since my binary_sensor.door_new_button_pressed of the VTO4202FB-P-S3 simply does not work with or without SIP server enabled on the VTO.

Anyway @felipecrs , thanks for the automation, and the other insights.... it was a huge help to finish my environment.

**Comment by felipecrs** [comment]
Great. This saves everyone from the pain of having to configure Asterisk.

I will also try it and update the guide accordingly if all works fine.

**Comment by pedromrg** [comment]
Also you might be able to play with this and answer the call after X seconds with a recorded message or.... need some more time to investigate...

**Comment by felipecrs** [comment]
I was able to play audio (like Google TTS) to the doorbell using Alexxit's WebRTC integration media player.

So, as long as the call is not ongoing, it can be used to play any audio back, including a message.

**Comment by thelittlechef** [comment]
Long time listener, first time caller.

Using [HA-SIP](https://github.com/felipecrs/dahua-vto-on-home-assistant/issues/1#issuecomment-2439040920) and the VTO acting as SIP server, as well as [VTO2MQTT](https://github.com/felipecrs/dahua-vto-on-home-assistant/issues/1#issuecomment-2374028000), I can now reliably* get MQTT messages when the button is pressed. I have a VTO 2211G-WP.

I had no luck at all with Asterisk. Entirely user error, I suspect, with TLS proving somewhat beyond my skills...

*Well - the VTO2MQTT container crashed and needed a "systemctl docker restart" to get going again, so maybe not so reliable...

**Comment by moskovskiy82** [comment]
@felipecrs Felipe you made a great work. I got everything up and running without the SIP server. It might be useful to update the docs as the SIP server you are using is only for catching the button pressed event. On some Dahua this can be done easily with the dahua integration in HA. This is what i've got in my automation as a trigger

```
  trigger:
  - platform: state
    entity_id: binary_sensor.kalitka_vto_button_pressed
    to: 'on'
  action:
```

**Comment by LiveBootleg** [comment]
First of all, Felipe - this is an absolutely wonderful piece of work, thanks for your efforts. I hope I can get past my specific issue and make it work the same way you have.

I have a Dahua Doorbell model DHI-VTO2211G-WP and initially integrated it with Home Assistant using both the Dahua and Dahua VTO integrations. When I initially installed the integration I did manage to get a doorbell.vto_button_pressed binary sensor and I could trigger a RF chime inside my house and it worked perfectly. I also have Frigate (HA add-on) and the video feed from the camera is working as expected.

I then went on to update the firmware on the Dahua VTO device (don't ask me why I did that) to version 2024-08-29 V4.700.0000000.2.R and suddenly the doorbell.vto_button_pressed binary sensor disappeared - not exactly sure what happened there. I can still interact with the device using the dahua_vto.send_command but I was not able to catch the doorbell button press in any other way than to use Asterisk. That is now working thanks to your guide and I can respond when someone presses the doorbell button.

Now I want to get the two-way audio working. I've replicated all the automations you have provided and they seem to work fine as well as the Advanced Camera Card. I don't get the restream working though. if I specify my doorbell path as a normal RTSP stream in Frigate like this it works fine:

`
rtsp://username:password@192.168.0.123:554/cam/realmonitor?channel=1&subtype=0&unicast=true&proto=Onvif
`

When I define the doorbell to use the go2rtc restreaming I lose the video feed in Frigate with error 'No frames have been received. Check error logs' and the following error appears in the Frigate logfile:

`
2025-03-09 09:21:12.064571929  11:21:12.064 WRN [rtsp] error="streams: fork/exec /config/scripts/fix_vto_codecs.sh: exec format error, read tcp 127.0.0.1:35998->127.0.0.1:8554: i/o timeout" stream=doorbell
`

The error occurs a number of times and the reference to port 35998 seems strange, every occurrence has a different port number though? I have placed the shell script in the folder /config/scripts/fix_vto_codecs.sh and gave it execution permission.

Any ideas what may be the issue?

**Comment by felipecrs** [comment]
Your issue is not related to Asterisk. If you want to keep discussing it, please open a separate one.

Have you tried calling the script by yourself to check if it works?

```console
$ /config/scripts/fix_vto_codecs.sh rtsp://username:password@192.168.0.123:554/cam/realmonitor?channel=1&subtype=0&unicast=true&proto=Onvif
```

It should output `rtsp://username:password@192.168.0.123:554/cam/realmonitor?channel=1&subtype=0&unicast=true&proto=Onvif` if it works.

**Comment by LiveBootleg** [comment]
Apologies, did not mean to hijack the thread. I was not able to open a new issue on Github.

I did try to execute the script myself and it gave an error with the & in the query string. I then figured that the VTO has https enabled and I also added the 554 port. My config no looks like this:

`
go2rtc:
  streams:
    doorbell:
      # H264
      - echo:/config/scripts/fix_vto_codecs.sh --https 'ffmpeg:rtsp://username:password@192.168.1.123:554/cam/realmonitor?channel=1&subtype=0#video=copy'
      # PCMA, 2-way audio
      - echo:/config/scripts/fix_vto_codecs.sh --https 'rtsp://username:password@192.168.1.123:554/cam/realmonitor?channel=1&subtype=0&unicast=true&proto=Onvif#media=audio#backchannel=1'
      # AAC
      - rtsp://127.0.0.1:8554/doorbell?audio=aac
`

Two-way audio now works - thank you so much for this...

**Comment by felipecrs** [comment]
Awesome!

**Comment by moskovskiy82** [comment]
Actually i would delete any mention on asterisk. Working like a charm without one

**Comment by felipecrs** [comment]
@moskovskiy82 do you have a Dahua indoor panel (VTH)?

**Comment by felipecrs** [comment]
Let me clarify: you don't need Asterisk if you have a VTH. Probably you don't need Asterisk if you enable Dahua Cloud P2P too.

**Comment by Cr4z33** [comment]
@felipecrs yeah, but it is my understanding that without your solution I won't be able to have 2-way comm into HA right?

I've seen some SIP methods, but they were a nightmare to setup and never worked for me.

**Comment by felipecrs** [comment]
Ok, let me clarify this too: Asterisk is only required so that the button pressed sensor works in the Dahua integration. It is not required for anything else, including 2 way audio.

**Comment by moskovskiy82** [comment]
> [@moskovskiy82](https://github.com/moskovskiy82) do you have a Dahua indoor panel (VTH)?

Only VTO3211D

**Comment by felipecrs** [comment]
How do you capture the button press event? Through the Dahua integration?

**Comment by moskovskiy82** [comment]
> How do you capture the button press event? Through the Dahua integration?

Exactly


```
triggers:
  - entity_id: binary_sensor.kalitka_vto_button_pressed
    to: "on"
    trigger: state
conditions: []
actions:
  - target:
      entity_id: input_boolean.kalitka_calling
    action: input_boolean.turn_on
    data: {}
  - target:
      entity_id: camera.kalitka_vto_main
    action: dahua.vto_cancel_call
    data: {}
  - delay:
      hours: 0
      minutes: 0
      seconds: 1
      milliseconds: 0
```

**Comment by felipecrs** [comment]
Man, I think you are lucky. Maybe your VTO is different than mine in this regard.

**Comment by Cr4z33** [comment]
@felipecrs I just got the other day my new **_DHI-VTO3311Q-WP_** video doorbell and two **_DH-VTH2621G-P_** and I can confirm too that with this hardware you don't need Asterisk.

Surfing into this doorbell's web UI I see way more options, but all I had to do was the generic setup (set doorbell as SIP server, set IP addresses, passwords and bind the two VTH to two extensions).

I didn't have to edit  a single line into Home Assistant also because I _recycled_ the same IP address and name for the new doorbell.

Now I only have to figure out how to have your solution  working for one VTH only because they were bought for two different apartments (on at my house at the ground floor and the other at the 1st floor) and it rings/notifies whatever of the two doorbell's buttons are pressed.

I guess it can be done with **_Dahua ConfigTools_** in the **_Building Config_** section, but so far it's quite complicated to me... 😖

**Comment by felipecrs** [comment]
That's great, I'm happy it's working for you.

My brother also has a two-button device, and the only way I found to detect the button press for each button was to use Asterisk, and making each button call different numbers which I can capture and trigger automations from Home Assistant side.

I would say this needs to be added as a new feature of the Dahua integration. I suggest you open an issue there if there isn't any already.

**Comment by Cr4z33** [comment]
@felipecrs I am not quite sure that's the first thing to do because I am aware that I should first (try to) setup the two VTHs for two different "buildings" by using Dahua ConfigTool and only if that fails I would then open an issue.

As I said earlier THERE IS a way to do it in that app, but I didn't get it working yet hehe. 😅

**Comment by felipecrs** [comment]
But even if you are able to configure the screens properly, the Dahua integration does not support differentiating between the two button presses.

But you're right, one step at a time. :)

**Comment by Cr4z33** [comment]
Just for the record if you want to learn how to properly set each VTH for a different apartment give a look at [this](https://www.youtube.com/watch?v=bi0L_Uu7RGY) video tutorial made by Dahua.

I can now have your project working for my extension only as I wanted, but I had to set the top first doorbell button for it.

The below other apartment button now doesn't ring in Home Assistant anymore so I finally got what I wanted!

**Comment by felipecrs** [comment]
That's really great. Lucky you don't care about the second button. :D

**Comment by felipecrs** [comment]
Looks like the only alternative after all is [ha-sip](https://github.com/arnonym/ha-plugins), but ha-sip is also an add-on like Asterisk. I don't see much appeal on swapping one for another.

**Comment by krzakx** [comment]
Hi guys! 
For mine VTO 2311R-WP the _pressed button_ works however I can not cancel the call ! :( 

I do use:

`action: dahua.vto_cancel_call
target:
  entity_id: camera.dahua_main
`

any idea for me? I have VTO owns SIP server enabled.

**Comment by felipecrs** [comment]
You should open an issue in https://github.com/rroller/dahua if there isn't one already.

But I'd recommend you check if it works when using Asterisk, like recommended in this project.
